templates/workloads/common/cf/autoscaling.yml

```
@@ -0,0 +1,58 @@
+{{- if .Autoscaling}}
+AutoScalingRole:
+  Type: AWS::IAM::Role
+  Properties:
+    RoleName: !Join ['', [!Ref WorkloadName, AutoScalingRole]]
+    AssumeRolePolicyDocument:
+      Statement:
+        - Effect: Allow
+          Principal:
+            Service: ecs-tasks.amazonaws.com
+          Action: 'sts:AssumeRole'
+    ManagedPolicyArns:
+      - 'arn:aws:iam::aws:policy/service-role/AmazonEC2ContainerServiceAutoscaleRole'
+
+AutoScalingTarget:
+  Type: AWS::ApplicationAutoScaling::ScalableTarget
+  Properties:
+    MinCapacity: {{.Autoscaling.MinCapacity}}
+    MaxCapacity: {{.Autoscaling.MaxCapacity}}
+    ResourceId:
+      Fn::Join:
+        - '/'
+        - - 'service'
+          - Fn::ImportValue:
+              !Sub '${AppName}-${EnvName}-ClusterId'
+          - !GetAtt Service.Name
+    ScalableDimension: ecs:service:DesiredCount
+    ServiceNamespace: ecs
+    RoleARN: !GetAtt AutoScalingRole.Arn
+{{if .Autoscaling.CPU}}
+AutoScalingPolicyECSServiceAverageCPUUtilization:
+  Type: AWS::ApplicationAutoScaling::ScalingPolicy
+  Properties:
+    PolicyName: !Join ['-', [!Ref WorkloadName, ECSServiceAverageCPUUtilization]]
+    PolicyType: TargetTrackingScaling
+    ScalingTargetId: !Ref AutoScalingTarget
+    TargetTrackingScalingPolicyConfiguration:
+      PredefinedMetricSpecification:
+        PredefinedMetricType: ECSServiceAverageCPUUtilization
+      ScaleInCooldown: 10
+      ScaleOutCooldown: 10
```

I believe the minimum number is 60 that we can set. Maybe for scaling in we can wait a little longer to make sure we don't prematurely scale in?
```suggestion
      ScaleInCooldown: 120
      ScaleOutCooldown: 60
```